packages/@dcl/ecs/src/engine/index.ts

```diff
@@ -196,6 +196,20 @@ export function Engine({ transports }: IEngineParams = {}): IEngine {
       }
     }
 
+    // Transform parenting: cyclic dependency checker
+    // It checks only in modified Transforms
+    for (const entity of baseComponents.Transform.dirtyIterator()) {
+      let transform = baseComponents.Transform.getOrNull(entity)
+      while (transform && transform.parent) {
+        if (transform.parent === entity) {
+          dcl.error(`There is a cyclic parent with entity ${entity}`)
+          break
+        } else {
+          transform = baseComponents.Transform.getOrNull(transform.parent)
+        }
+      }
+    }
```

Comment:
I think there will be a tremendous penalty cost associated with this checks, can seting a "getter/setter" to the transform help reduce that number? should we do it only in dev mode instead? and not run the validation in production code?

Reply:
This `for iteration` is a system, and even it might be an optional system for debugger purposes.